### PR TITLE
Fixed issue of REST API Customer Create API & Customer Token Integration API

### PR DESCRIPTION
**Pull request by hiren0241**
Fixed issue of REST API Customer Create API & Customer Token Integration API

### Description (*)
This pull request fixed issue of customer REST API by checking store id. 

### Fixed Issues (if relevant)
1. magento/magento2#18088: Issue with REST API Customer Create API & Customer Token Integration API

### Manual testing scenarios (*)
1. Create customer through API
2. Get customer token through API

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @hiren0241. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by larsroettig — CHANGES_REQUESTED** [review]
Please cover this with an API Integrations Test

**Comment by larsroettig** [comment]
Hi@hiren0241,
thank you for contributions. Two things we don't accept direct PR´s on **2.2** please open this PR on Branch **magento:2.3-develop**. Then you can Backport this Change to **2.2**. 

Also please cover this PR with an  API Integrations Test. 


best regards,

Lars

**Comment by hiren0241** [comment]
HI @larsroettig,

Thank you for your valuable feedback and review.

I have gone through AccountManagement.php file on 2.3 develop branch and found that this file is already modify in that branch. Please refer this  commit: https://github.com/magento/magento2/commit/c51436560efbb5aa930dd50547a43b8922929752.  Someone has fixed this stuff over there for another issue. But, I am unable to find his appropriate PR. So, I have created PR in 2.2 develop branch. Is it possible for you to give me that particular PR id for 2.3 develop branch? so that, I can backport it or what is the other possible way to link this stuff over there. 

Please guide me for the same.

**Comment by larsroettig** [comment]
Hi @hiren0241, I ask in the Maintainers Team what we do in this case. It takes some days before we can respond to your questions. 

Best regards, 

Lars

**Comment by hiren0241** [comment]
Hi @larsroettig ,

Thanks for your quick response. I will wait till that time and I will do further changes after your response or maintainer's reply.

Thank you.

**Review by orlangur — CHANGES_REQUESTED** [review]
Hi @hiren0241, you need to cherry-pick corresponding commits like https://github.com/magento/magento2/commit/e3940f0de98155da20329988d5dd96d010ca3e27 manually from `2.3-develop`.

Here is an easy way to find appropriate commits: https://github.com/magento/magento2/search?q=MAGETWO-96125&type=Commits

**Comment by hiren0241** [comment]
Hi @orlangur,

I did the requested changes. Please review it

Thanks.

**Comment by orlangur** [comment]
@hiren0241 please get rid of unnecessary commits, like https://github.com/magento/magento2/pull/19662/commits/5c8dd646e5604d85a0ae9ceee5e51bbf981fe279. There should be only cherry-picked commits and then additional commit authored by you on top of them if needed.

**Comment by hiren0241** [comment]
Hi @orlangur,

Okay, I will do the same. Thanks for reviewing it.

**Comment by hiren0241** [comment]
Hi @orlangur,

I did the requested changes. Please review it once again.

Thanks.

**Comment by sidolov** [comment]
Hi @hiren0241 , please, take a look at merge conflicts. 
thank you!

File: app/code/Magento/Customer/Api/AccountManagementInterface.php [hunk 1]
```diff
@@ -31,13 +31,15 @@ interface AccountManagementInterface
...
+     * @param array $extensions
      * @return \Magento\Customer\Api\Data\CustomerInterface
      * @throws \Magento\Framework\Exception\LocalizedException
      */
     public function createAccount(
         \Magento\Customer\Api\Data\CustomerInterface $customer,
         $password = null,
-        $redirectUrl = ''
+        $redirectUrl = '',
+        $extensions = []
```
Comment (sidolov) [inline app/code/Magento/Customer/Api/AccountManagementInterface.php:42]: Unfortunately we can't change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/), especially on the API interfaces, we can't modify them at all.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sivaschenko** [comment]
Closing this pull request due to inactivity, feel free to reopen if you'd like to continue progress

**Pull request closed without merging**